src/components/structures/login/Login.js

```diff
@@ -74,6 +77,9 @@ module.exports = React.createClass({
     componentWillMount: function() {
         this._unmounted = false;
         this._initLoginLogic();
+        this.setState({
+            language: languageHandler.getCurrentLanguage(),
```

Comment:
can we not do this in `getInitialState` ?

and if not, please can you set `language: null` in `getInitialState`, so that we have a list of all the `state` fields that get set in one place?